pkg/iphelpers/iphelpers.go

```diff
@@ -25,22 +25,27 @@ func CompareIPs(ipX net.IP, ipY net.IP) int {
 	return 0
 }
 
-// IPGetOffset gets offset between ip1 and ip2. This assumes ip1 > ip2 (from IP representation point of view)
-func IPGetOffset(ip1, ip2 net.IP) uint64 {
+// IPGetOffset gets the absolute offset between ip1 and ip2, meaning that this offset will always be a positive number.
+func IPGetOffset(ip1, ip2 net.IP) (uint64, error) {
 	if ip1.To4() == nil && ip2.To4() != nil {
-		return 0
+		return 0, fmt.Errorf("cannot calculate offset between IPv4 (%s) and IPv6 address (%s)", ip1, ip2)
```

Comment:
Good catch. The To16 though doesn't work, because an IPv4 address in 16 byte notation can still be compared to an IPv4 address in 4 byte notation (they are both converted to 16 bytes a few lines below)